quality_check/11-rnaseq_bias.Rmd

```diff
@@ -0,0 +1,294 @@
+  # (e.g., SRPXXXXXX) for an RNA-seq sample
+  # 
+  # Args:
+  #   accession_code: sample accession code (should be RNA-seq!)
+  #
+  # Returns:
+  #   project_accession: the project accession code associated with sample,
+  #                      which is extracted from the European Nucleotide 
+  #                      Archive reference URL
+  
+  sample_info <- refinebio_sample_code_to_list(accession_code)
+  
+  # this is only intended to work for RNA-seq samples
+  if(sample_info$results$technology != "RNA-SEQ") 
+    stop("get_sample_project_accession only works with RNA-seq samples!")
+  
+  # what project accession number did this come from? we can get this from the
+  # url of the source reference stored in refine.bio
+  project_accession <- NA
+  reference_url <- sample_info$results$protocol_info[[1]]$Reference
+  if (!is.null(reference_url))
+    project_accession <- stringr::word(reference_url, -1, sep = "/")
+  return(project_accession)
+}
+```
+
+```{r}
+extract_sample_metadata <- function(accession_code) {
+  # given an RNA-seq sample (run) accession code (e.g., SRRXXXXX), return the 
+  # following metadata from the refine.bio API as a data.frame that is a 
+  # single row: 
+  #   sample accession code, project accession code (e.g.), library layout, 
+  #   percent of reads mapped by salmon quant, number of reads, inferred library
+  #   type, compatible fragment ratio, which length salmon index was used
+  # 
+  # Args:
+  #   accession_code: the sample (run in SRA parlance) accession code
+  #                   for an RNA-seq sample such as "SRR998890" (character)
+  # 
+  # Returns:
+  #   a data.frame of dimensions 1, 9 that contains the information for the
+  #   sample outlined above
+  #   
+  
+  # internal helper function -- there are a bunch of variables that we need
+  # to pull from a data.frame and get rid of multiple NAs so we'll use 
+  # rlang::sym to help us with that
+  extract_var_na_rm <- function(var_name = "percent_mapped") {
+    sample_info_df %>%
+      dplyr::filter(!is.na(!!rlang::sym(var_name))) %>%
+      dplyr::pull(!!rlang::sym(var_name))
+  }
+  
+  # get the sample annotations, etc. from the refine.bio API
+  sample_info <- refinebio_sample_code_to_list(accession_code)
+  
+  # this is only intended to work for RNA-seq samples
+  if(sample_info$results$technology != "RNA-SEQ") 
+    stop("extract_sample_metadata only works with RNA-seq samples!")
+  
+  # what's the SRP|ERP|DRP ? this will help us track down 
+  project_accession <- get_sample_project_accession(accession_code)
+  
+  # the sample annotations contain all the relevant information about the salmon
+  # run
+  sample_annotations <- sample_info$results$results[[1]]$annotations
+  
+  # there are a bunch of empty data.frame -- we don't want that! so let's find
+  # the index of what we want and extract it
+  nonempty_index<- which(unlist(lapply(lapply(sample_annotations, colnames), 
```

Comment:
This is a lot to follow on one line. Is there a way you can break it up? Or would that not be in accordance with the style guide?

Reply:
What we get out of `jsonlite::fromJSON` is a bit of a _messy mess_ so I don't know that I can make it much better honestly.